### PR TITLE
Fix type assertion for primitive type

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Fix type assertion for primitive type since we can't rely on a strict equality in that case.

### CODE REVIEW

File: nexus-sdk/src/main/java/io/nexusrpc/handler/ReflectionUtil.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+   * @return the boxed wrapper type for the give {@code type}, or {@code type} if no wrapper class
+   *     was found.
+   */
+  static Type wrapTypeIfPrimitive(Type type) {
+    Class<?> clazz = primitiveWrapperTypeMap.get(type);
+    if (clazz != null) {
+      return clazz;
+    }
+    return type;
+  }
```
Comment (cretz) [inline nexus-sdk/src/main/java/io/nexusrpc/handler/ReflectionUtil.java:36]: Can make the map value type a `Type` too and do:

```suggestion
  static Type wrapTypeIfPrimitive(Type type) {
    primitiveWrapperTypeMap.getOrDefault(type, type);
  }
```

But it's pedantic

File: nexus-sdk/src/main/java/io/nexusrpc/handler/ReflectionUtil.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package io.nexusrpc.handler;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+class ReflectionUtil {
+  /** A map of Primitive types to their respective wrapper types. */
+  private static final Map<Type, Class<?>> primitiveWrapperTypeMap = new HashMap<>(8);
```
Comment (cretz) [inline nexus-sdk/src/main/java/io/nexusrpc/handler/ReflectionUtil.java:9]:
```suggestion
  private static final Map<Type, Class<?>> primitiveWrapperTypeMap = new HashMap<>(9);
```

**Review by cretz — COMMENTED** [review]
Unfortunately Java reflection doesn't have this utility, so this LGTM, only minor suggestions

File: nexus-sdk/src/main/java/io/nexusrpc/handler/ReflectionUtil.java (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply nexus-sdk/src/main/java/io/nexusrpc/handler/ReflectionUtil.java]: Done

**Review by cretz — APPROVED** [review]

**Pull request merged**